ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml

```diff
@@ -70,293 +178,456 @@ StatusDialog {
-        }
-    }
 
-    function isFormValid() {
-        return (nameInput.valid &&
-                descriptionTextArea.valid) &&
-                Utils.validateAndReturnError(colorDialog.color.toString().toUpperCase(),
-                                             communityColorValidator) === ""
+        updateRightButtons()
     }
 
-    StatusScrollView {
-        id: scrollView
-
-        property ScrollBar vScrollBar: ScrollBar.vertical
-
-        anchors.fill: parent
-        contentWidth: availableWidth
-        padding: 16
-
-        function scrollBackUp() {
-            vScrollBar.setPosition(0)
-        }
+    width: 640
 
+    readonly property list<Item> discordPages: [
         ColumnLayout {
-            id: content
-
-            width: scrollView.availableWidth
-            spacing: 0
+            id: fileListView
+            objectName: "discordFileListView" // !!! DON'T CHANGE, clearFilesButton depends on this
+            spacing: 24
+            readonly property var fileListModel: root.communitiesStore.discordFileList
+            readonly property bool fileListModelEmpty: !fileListModel.count
+
+            readonly property bool canGoNext: fileListModel.selectedCount
+                                              || (fileListModel.selectedCount && fileListModel.selectedFilesValid)
+            readonly property string nextButtonText: fileListModel.selectedCount && fileListModel.selectedFilesValid ?
+                                                         qsTr("Proceed with (%1/%2) files").arg(fileListModel.selectedCount).arg(fileListModel.count) :
+                                                         fileListModel.selectedCount ? qsTr("Validate (%1/%2) files").arg(fileListModel.selectedCount).arg(fileListModel.count)
+                                                                                     : qsTr("Start channel import")
+            readonly property var nextAction: function () {
+                if (!fileListView.fileListModel.selectedFilesValid) {
+                    return root.communitiesStore.requestExtractChannelsAndCategories()
+                }
+                root.currentIndex++
+            }
 
-            StatusInput {
-                id: nameInput
+            RowLayout {
                 Layout.fillWidth: true
-                input.edit.objectName: "createOrEditCommunityChannelNameInput"
-                label: qsTr("Channel name")
-                charLimit: root.maxChannelNameLength
-                placeholderText: qsTr("# Name the channel")
-
-                input.onTextChanged: {
-                    const cursorPosition = input.cursorPosition
-                    input.text = Utils.convertSpacesToDashes(input.text)
-                    input.cursorPosition = cursorPosition
-                    if (root.channelEmoji === "") {
-                        input.letterIconName = text;
+                spacing: 12
+                StatusBaseText {
```

Comment:
Texts like that should be prepared for input longer then initial English text. Even with slightly longer text positioning is broken here.

Reply:
Just replace "Select Discord channel JSON files to import" text with a bit longer text. The layout is broken then. It may easily happen for translations where given sentence is longer than in English.

Reply:
Ah I see! Will add an elide/wrap there

Reply:
Fixed @micieslak, thanks!